solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -448,30 +448,68 @@ library DepositUtils {
     /// @param _redeemer    The assumed owner of the deposit's TDT.
     /// @return             The amount in TBTC needed to redeem the deposit.
     function getOwnerRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns(uint256) {
-        uint256 fee = signerFee(_d);
-        bool inCourtesy = _d.inCourtesyCall();
-        if(remainingTerm(_d) > 0 && !inCourtesy){
-            if(feeRebateTokenHolder(_d) != _redeemer) {
-                return fee;
-            }
-        }
-        uint256 contractTbtcBalance = _d.tbtcToken.balanceOf(address(this));
-        if(contractTbtcBalance < fee) {
-            return fee.sub(contractTbtcBalance);
-        }
-        return 0;
+        uint256 frtAdjust = frtAdjust(_d, _redeemer);
+        uint256 signerFeeAdjust = signerFeeAdjust(_d, _redeemer);
+        return frtAdjust.add(signerFeeAdjust);
     }
 
     /// @notice             Get TBTC amount required by redemption by a specified _redeemer.
     /// @dev                Will revert if redemption is not possible by msg.sender.
     /// @param _redeemer    The deposit redeemer.
     /// @return             The amount in TBTC needed to redeem the deposit.
     function getRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns(uint256) {
+        uint256 mainCharge = mainCharge(_d, _redeemer);
+        uint256 frtAdjust = frtAdjust(_d, _redeemer);
+        uint256 signerFeeAdjust = signerFeeAdjust(_d, _redeemer);
+        return mainCharge.add(frtAdjust).add(signerFeeAdjust);
+    }
+
+    /// @notice             Get the base TBTC amount needed to redeem.
+    /// @param _redeemer    The deposit redeemer.
+    /// @return             The amount in TBTC.
+    function mainCharge(DepositUtils.Deposit storage _d, address _redeemer) internal view returns (uint256){
```

Comment:
I thought of doing exactly this, but decided against it since I thought it would just be more code. On second thought it's really not too bad, and does make things a little more clear